tests/src/test/scala/common/WskTracingTests.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package common
+
+import io.opentracing.Span
+import io.opentracing.mock.{MockSpan, MockTracer}
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import pureconfig.loadConfigOrThrow
+import whisk.common.{LoggingMarkers, TransactionId}
+import whisk.common.tracing.{OpenTracer, TracingConfig}
+import whisk.core.ConfigKeys
+
+import scala.ref.WeakReference
+
+@RunWith(classOf[JUnitRunner])
+class WskTracingTests extends TestHelpers {
+
+  val tracer: MockTracer = new MockTracer()
+  val sleepTime = 10
+  val tracingConfig = loadConfigOrThrow[TracingConfig](ConfigKeys.tracing)
+  val openTracer = new OpenTracer(tracer, tracingConfig)
+
+  it should "create span and context and invalidate cache after expiry" in {
+    tracer.reset
+    val transactionId: TransactionId = TransactionId.testing
+    var list: List[WeakReference[Span]] = List()
+
+    openTracer.startSpan(LoggingMarkers.CONTROLLER_ACTIVATION, transactionId)
+    var ctx = openTracer.getTraceContext(transactionId)
+    openTracer.setTraceContext(transactionId, ctx)
+    assert(ctx.isDefined)
+
+    Thread.sleep((tracingConfig.cacheExpiry.toMillis + 5000))
+    ctx = openTracer.getTraceContext(transactionId)
+    assert(!ctx.isDefined)
+    openTracer.startSpan(LoggingMarkers.CONTROLLER_KAFKA, transactionId)
+    Thread.sleep(sleepTime)
+    openTracer.finishSpan(transactionId)
+    val finishedSpans = tracer.finishedSpans()
+    assert(finishedSpans.size() == 1)
```

Comment:
Can you rewrite this to `finishedSpans should have size 1` (might need to convert the collection to scala maybe). That would yield a debuggable message upon failure.